mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -59,22 +60,47 @@ void strToTokens(const std::string &arguments,
 
 } // namespace
 
-int Conv2dGenerator::getKernelCount() const {
-  int count = 0;
+llvm::SmallVector<int> Conv2dGenerator::getKernelCount() const {
+  llvm::SmallVector<int> gemmIds;
   if (config.kernelId > 0) { // generate only 1 specified kernel
-    count = 1;
+    gemmIds.push_back(0);
   } else if (config.operation == "conv2d") {
-    count = 1;
+    gemmIds.push_back(0);
   } else if (config.operation == "conv2d_bwd_data") {
-    count = 1;
+    gemmIds = getBwdDataKernelCount();
   } else if (config.operation == "conv2d_bwd_weight") {
-    count = 1;
+    gemmIds.push_back(0);
   } else if (config.operation == "conv2d_dummy") {
-    count = 1;
+    gemmIds.push_back(0);
   }
-  return count;
+  return gemmIds;
 }
+// calculate gemmId of gemm that can be runned
```

Comment:
The comment needs to be elaborated. One crucial observation is NOT all numbers from `0..n-1` would be populated in the vector when `n` kernels are potentially needed.